robotframework_reportportal/model.py

```diff
@@ -83,7 +83,7 @@ def __init__(self, name, attributes):
         self.code_ref = '{0}:{1}'.format(attributes['source'], name)
         # for backward compatibility with Robot < 4.0 mark every test case
         # as critical if not set
-        self.critical = attributes.get('critical', 'yes') == 'yes'
+        self.critical = attributes.get('critical', True)
```

Comment:
Legacy Robot versions set `yes` or `no` on this attribute. That's why we need to compare it with `yes` to get a pure `bool` value. Further we use is like here:
```python
if not test.critical and test.status == 'FAIL':
    test.status = 'SKIP'
```
AFAIK your change will break this logic, since literal 'no' will be treated as positive value and the condition will never happen.

Reply:
It's not correct from what I see in the Robot Framework [source code](https://robot-framework.readthedocs.io/en/v3.0.4/_modules/robot/result/model.html#TestCase).

```
    @property
    def critical(self):
        """``True/False`` depending on is the test considered critical.

        Criticality is determined based on test's :attr:`tags` and
        :attr:`~TestSuite.criticality` of the :attr:`parent` suite.
        """
        if not self.parent:
            return True
        return self.parent.criticality.test_is_critical(self)
```